ddtrace/tracer.py

```diff
@@ -41,6 +53,59 @@
         logging.basicConfig(level=logging.DEBUG)
 
 
+@attr.s()
+class _Trace(object):
+    trace_id = attr.ib(type=int)  # type: int
+    sampled = attr.ib(type=bool, default=True)  # type: bool
+    num_finished = attr.ib(type=int, default=0)  # type: int
+    sampling_priority = attr.ib(default=None)  # type: Optional[int]
+    dd_origin = attr.ib(default=None)  # type: Optional[str]
+    local_root_span = attr.ib(default=None)  # type: Optional[Span]
+    _spans = attr.ib(default=attr.Factory(list))  # type: List[Span]
+
+    def __len__(self):
+        return len(self._spans)
+
+    def add_span(self, span):
+        # type: (Span) -> None
+        self._spans.append(span)
+
+    def pop_finished_spans(self):
+        # type: () -> List[Span]
+        local_root_span = self.local_root_span
+        if local_root_span:
+            if self.sampling_priority is not None and self.sampled:
+                local_root_span.set_metric(SAMPLING_PRIORITY_KEY, self.sampling_priority)
+            if self.dd_origin:
+                local_root_span.meta[ORIGIN_KEY] = str(self.dd_origin)
+
+        finished_spans = [span for span in self._spans if span.finished]
+        self._spans = [span for span in self._spans if not span.finished]
+        self.num_finished -= len(finished_spans)
+        return finished_spans
+
+
+_traces = {}  # type: Dict[int, _Trace]
```

Comment:
this could be friendlier encapsulated in a class.

```python
class Traces:
    _traces = {}

    def get_trace(self, trace_id):
        return self._traces.get(trace_id)


traces = Traces()
```

No strong opinion here, assuming you already thought of this anyways.